doc/source/guide/in_depth/functional_guide.rst

```diff
@@ -0,0 +1,257 @@
+   However, Hilbert spaces are self-dual, meaning :math:`X^* = X`, and :math:`\langle x^*,x \rangle` is then the inner product.
+
+The optional method is:
+
+ * ``proximal(sigma)``. This returns the proximal operator of the functional, where ``sigma`` is a nonnegative step-size like parameter.
+   The proximal operator is defined as
+
+   .. math::
+
+      \text{prox}_{\sigma f}(x) = \text{arg min}_{y \in X} \{ f(y) + \frac{1}{2\sigma} \|y - x\|_2^2 \}. 
+
+The `Functional` class also contains two default implementations of two help functions:
+
+ * ``derivative(point)``. Given an implementation of the gradient, this method return the (directional) derivative operator in ``point``.
+   This is the linear operator
+
+   .. math::
+
+      x \to \langle x, \nabla f(point) \rangle,
+
+   where :math:`\nabla f(point)` is the gradient of the functional in the point :math:`point`.
+
+ * ``translate(shift)``. Give a functional :math:`f(.)`, this method creates the functional :math:`f(. - shift)`
+
+
+Functional arithmetics
+======================
+It is common in applications to perform arithmetic operations with functionals, for example adding two functionals :math:`f` and :math:`g`
+
+.. math::
+   [f+g](x) = f(x) + g(x),
+
+or multiplication of a functional by a scalar
+
+.. math::
+   [\alpha f](x) = \alpha f (x).
+
+Another example is translating a functional with a vecotr :math:`y`
+
+.. math::
+   f(x - y),
+
+or given an `Operator` :math:`A` whose range is the same as the domain of the functional we also have composition
+
+.. math::
+    [f * A](x) = f(A(x)). 
+
+In some of these cases, properties and methods such as ``gradient``, ``convex_conjugate`` and ``proximal`` can be calculated automatically given a default implementation of the corresponding property in :math:`f`.
+
+All available functional arithmetic, including which properties and methods that automatically can be calculated, is shown below. ``f``, ``g`` represent `Functional`'s, and ``A`` an `Operator` whose range is the same as the domain as the functional. ``a`` is a scalar in the field of the domain of ``f`` and ``g``, and ``y`` is a vector in the domain of ``f`` and ``g``.
+
++--------------------+-----------------+-------------------------------------------------+
+| Code               | Meaning         | Class                                           |
++====================+=================+=================================================+
+| ``(f + g)(x)``     | ``f(x) + g(x)`` | `FunctionalSum`                                 |
+|                    |                 | - Retains `gradient`.                           |
++--------------------+-----------------+-------------------------------------------------+
+| ``(f + a)(x)``     | ``f(x) + a``    | `FunctionalScalarSum`                           |
```

Comment:
`S + a` could be ambiguous if the functional `S` is defined on the real numbers. Boring case for sure, but perhaps worth mentioning that it never means "constant times input".